Trigger/src/EvalWeightedEventCounts_module.cc

```diff
@@ -0,0 +1,223 @@
+
+#include <cmath>
+// #include <iostream>
+#include <string>
+// #include <map>
+#include <vector>
+
+namespace mu2e {
+
+class EvalWeightedEventCounts : public art::EDAnalyzer {
+
+public:
+  using Name = fhicl::Name;
+  using Comment = fhicl::Comment;
+  struct Config {
+    fhicl::Atom<int> diagLevel{Name("diagLevel"), Comment("Debug Level"), 0};
+    fhicl::Atom<art::InputTag> evtWeightTag{
+        Name("protonBunchIntensity"), Comment("Proton Bunch Intenity "), "protonBunchIntensity"};
+    fhicl::Atom<float> nProcess{Name("nEventsProcessed"), Comment("Events Processed"), 1.};
+    fhicl::Atom<std::string> tagName{Name("tagName"), Comment("Tag Name"), "tagName"};
+  };
+
+  using Parameters = art::EDAnalyzer::Table<Config>;
+
+  enum {
+    kNTrigInfo = 40,
+    kNTrackTrig = 20,
+    kNTrackTrigVar = 30,
+    kNHelixTrig = 40,
+    kNHelixTrigVar = 130,
+    kNCaloCalib = 5,
+    kNCaloCalibVar = 30,
+    kNCaloOnly = 5,
+    kNCaloOnlyVar = 30,
+    kNOcc = 100,
+    kNOccVar = 100
+  };
+
+  struct occupancyHist_ {
+    TH1F* _hOccInfo[kNOcc][kNOccVar];
+    TH2F* _h2DOccInfo[kNOcc][kNOccVar];
+
+    occupancyHist_() {
+      for (int i = 0; i < kNOcc; ++i) {
+        for (int j = 0; j < kNOccVar; ++j) {
+          _hOccInfo[i][j] = NULL;
+          _h2DOccInfo[i][j] = NULL;
+        }
+      }
+    }
+  };
+
+  explicit EvalWeightedEventCounts(const Parameters& config);
+  virtual ~EvalWeightedEventCounts() {}
+
+  virtual void beginJob();
+  virtual void endJob();
+
+  // This is called for each event.
+  virtual void analyze(const art::Event& e);
+  virtual void beginRun(const art::Run& run);
+
+  void bookHistograms();
+  void bookOccupancyInfoHist(art::ServiceHandle<art::TFileService>& Tfs, occupancyHist_& Hist);
+
+private:
+  int _diagLevel;
+  art::InputTag _evtWeightTag;
+
+  float _nProcess;
+  std::string _tagName;
+  double _nPOT;
+
+  occupancyHist_ _occupancyHist;
+
+  const mu2e::Tracker* _tracker;
+
+  // the following pointer is needed to navigate the MC truth info of the strawHits
+  const art::Event* _event;
+
+  float _minPOT, _maxPOT;
+  double _wgSum[3] = {0};
+};
+
+EvalWeightedEventCounts::EvalWeightedEventCounts(const Parameters& config) :
+    art::EDAnalyzer(config), _diagLevel(config().diagLevel()), _evtWeightTag(config().evtWeightTag()),
+    _nProcess(config().nProcess()), _tagName(config().tagName()), _minPOT(1e6), _maxPOT(4e8) {}
+
+void EvalWeightedEventCounts::bookHistograms() {
+  art::ServiceHandle<art::TFileService> tfs;
+
+  bookOccupancyInfoHist(tfs, _occupancyHist);
+}
+
+void EvalWeightedEventCounts::bookOccupancyInfoHist(art::ServiceHandle<art::TFileService>& Tfs,
+                                                    occupancyHist_& Hist) {
+
+  int index_last = 0;
+  art::TFileDirectory occInfoDir = Tfs->mkdir("occInfoGeneral");
+  Hist._hOccInfo[index_last][0] = occInfoDir.make<TH1F>(
+      Form("hInstLum_%i", index_last), "distrbution of instantaneous lum; p/#mu-bunch", 1000,
```

Comment:
Some time ago we decided to stop saying "micro bunch" and instead say "pulse".   The old terminology is still in wide use but we are changing to the new terminology as targets of opportunity.  So please change that in your histogram titles.

Reply:
Corrected to "pulse".